src/main/java/io/spine/text/TextFactory.java

```diff
@@ -0,0 +1,162 @@
+import static io.spine.util.Exceptions.newIllegalArgumentException;
+import static java.lang.System.lineSeparator;
+
+/**
+ * Static factories and precondition checks for creating instances of {@link Text}.
+ *
+ * @apiNote A recommended way for using this class is using its methods statically imported,
+ *         so that the creation of {@link Text} objects looks compact:
+ *         <pre>{@code
+ *         import io.spine.text.TextFactory.text
+ *         ...
+ *         var twoLines = text("one", "two");
+ *         }</pre>
+ */
+public final class TextFactory {
+
+    private static final Splitter SPLITTER = Splitter.on(lineSeparator());
+    private static final Joiner JOINER = Joiner.on(lineSeparator());
+    private static final Position NOT_FOUND = Position.newBuilder().setLine(-1).build();
+
+    /**
+     * Prevents instantiation of this static factory class.
+     */
+    private TextFactory() {
+    }
+
+    /**
+     * Creates a new instance with the given value.
+     */
+    public static Text text(String value) {
+        checkNotNull(value);
+        return Text.newBuilder()
+                .setValue(value)
+                .build();
+    }
+
+    /**
+     * Creates a new instance of text with lines separated by {@linkplain System#lineSeparator()
+     * line separator}.
+     */
+    public static Text text(Iterable<String> lines) {
+        checkNotNull(lines);
+        checkNoSeparators(lines);
+        var joined = JOINER.join(lines);
+        return text(joined);
+    }
+
+    /**
+     * Creates a new instance of text with lines separated by {@linkplain System#lineSeparator()
+     * line separator}.
+     */
+    public static Text text(String[] lines) {
+        checkNotNull(lines);
+        return text(ImmutableList.copyOf(lines));
+    }
+
+    /**
+     * Creates a new list with the given lines.
+     *
+     * @throws IllegalArgumentException
+     *          if any of the lines contains the {@linkplain System#lineSeparator()
+     *          line separator}
+     */
+    @VisibleForTesting
+    public static Text createText(String... lines) {
+        checkNotNull(lines);
+        return text(lines);
+    }
+
+    /**
+     * Ensures that lines do not contain {@linkplain System#lineSeparator()
+     * line separators}.
+     *
+     * @throws IllegalArgumentException
+     *          if at least one line contains a {@linkplain System#lineSeparator()
+     *          line separator}
+     */
+    public static void checkNoSeparators(Iterable<String> lines) {
+        lines.forEach(TextFactory::checkNoSeparator);
+    }
+
+    /**
+     * Ensures that charter sequence does not contain a {@linkplain System#lineSeparator()
+     * line separator}.
+     *
+     * @throws IllegalArgumentException
+     *          if the sequence contains a {@linkplain System#lineSeparator()
+     *          line separator}
+     */
+    public static void checkNoSeparator(CharSequence s) {
```

Comment:
I think `check...` functions belong in `Text` itself, if they truly need to be public.

Reply:
You're right. But since `Text` is a generated type adding static methods to it would take some time. 

I was thinking of using this small library also as an example of extended code generation with static factories and other methods added to the `Text` type. 

But let's fix the matter of existence of this library at all first. We planned to discuss ProtoData and Validation, and this library is one of the topics.